ddl/db_integration_test.go

```diff
@@ -2173,8 +2177,8 @@ func (s *testIntegrationSuite7) TestCreateExpressionIndexError(c *C) {
 	tk.MustExec("drop table if exists t;")
 	tk.MustExec("create table t (a int, b real);")
 	tk.MustGetErrCode("alter table t add primary key ((a+b));", errno.ErrFunctionalIndexPrimaryKey)
-	tk.MustGetErrCode("alter table t add index ((rand()));", errno.ErrGeneratedColumnFunctionIsNotAllowed)
-	tk.MustGetErrCode("alter table t add index ((now()+1));", errno.ErrGeneratedColumnFunctionIsNotAllowed)
+	tk.MustGetErrCode("alter table t add index ((rand()));", errno.ErrFunctionalIndexFunctionIsNotAllowed)
+	tk.MustGetErrCode("alter table t add index ((now()+1));", errno.ErrFunctionalIndexFunctionIsNotAllowed)
```

Comment:
Are these functions allowed in generated columns? Please add similar tests for generated columns.

Reply:
Yes, in test TestSqlFunctionsInGeneratedColumns (https://github.com/pingcap/tidb/blob/bcf9ffe3f6dfedeb969ac768b532ac006c5e95bc/ddl/db_integration_test.go#L2048) there are some cases for generated columns. Thus similar tests for generated functional indexes are just added.